src/models/frequencymodel.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+  frequencymodel.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "frequencymodel.h"
+
+FrequencyModel::FrequencyModel(QObject* parent)
+    : QAbstractListModel(parent)
+{
+}
+
+int FrequencyModel::rowCount(const QModelIndex& parent) const
+{
+    return m_results.data[parent.column() / 2].size();
+}
+
+int FrequencyModel::columnCount(const QModelIndex& /*parent*/) const
+{
+    return m_results.data.size() * 2;
+}
+
+QVariant FrequencyModel::data(const QModelIndex& index, int role) const
+{
+    if (role == Qt::DisplayRole) {
```

Comment:
reduce indent depth by first checking things and returning early otherwise

```
if (role != Qt::DisplayRole || !hasIndex(index.row(), index.column(), index.parent()) {
    return {};
}
const auto &data = m_results.data[index.column() / 2][index.row()];

if (index.column() % 2 == 0) {
    return data.time;
} else {
    return data.cost;
}
```